From bef6467cfbd8892acb6d71236ec0d54fe16c5c2a Mon Sep 17 00:00:00 2001
From: Han Gao <gaohan@iscas.ac.cn>
Date: Mon, 11 Nov 2024 23:30:40 +0800
Subject: [PATCH 348/540] fix:HACK: reject sscofpmf for kvm

Changelogs:
HACK: reject sscofpmf for kvm

cannot start VM properly after enabling
temporary reject probe sscofpmf

Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
Signed-off-by: linmin <linmin@eswincomputing.com>
---
 arch/riscv/kvm/vcpu_pmu.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/riscv/kvm/vcpu_pmu.c b/arch/riscv/kvm/vcpu_pmu.c
index e2e2a115afb5c..0a8ab8c3700fe 100644
--- a/arch/riscv/kvm/vcpu_pmu.c
+++ b/arch/riscv/kvm/vcpu_pmu.c
@@ -556,6 +556,8 @@ void kvm_riscv_vcpu_pmu_init(struct kvm_vcpu *vcpu)
 	 */
 	if (!riscv_isa_extension_available(NULL, SSCOFPMF))
 		return;
+	/* reject sscofpmf */
+	return;
 
 	ret = riscv_pmu_get_hpm_info(&hpm_width, &num_hw_ctrs);
 	if (ret < 0 || !hpm_width || !num_hw_ctrs)
-- 
2.53.0

